src/shared/tests.c

```diff
@@ -31,3 +34,13 @@ char* setup_fake_runtime_dir(void) {
 
         return p;
 }
+
+const char* get_exe_relative_testdata_dir(void) {
+        _cleanup_free_ char *exedir = NULL;
+        /* convenience: caller does not need to free result */
+        static char testdir[PATH_MAX];
+
+        assert_se(readlink_and_make_absolute("/proc/self/exe", &exedir) >= 0);
+        assert_se(snprintf(testdir, sizeof(testdir), "%s/testdata", dirname(exedir)) > 0);
```

Comment:
Not sure I understand how does it work
```
./libtool --mode=execute ./test-engine
...
Assertion 'r == 0' failed at src/test/test-engine.c:62, function main(). Aborting.
Aborted (core dumped)
```
`strace` shows:
```
stat("/home/vagrant/systemd/.libs/testdata", 0x7ffd852cece0) = -1 ENOENT (No such file or directory)
```
i.e. `make valgrind-tests` is broken